### PR TITLE
kromium 0.1.7 (new formula)

### PR DESCRIPTION
**Pull request by sharvanath**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/kromium.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  homepage "https://kromium.io"
+  url "https://github.com/sharvanath/kromium/archive/refs/tags/v0.1.7.tar.gz"
+  sha256 "5634282e989923b0f6960102d63f7e8fb6d91010ac36c6d4cf1fa8bc1e158eb5"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-o", "kromium"
+    bin.install "kromium"
```
Comment (carlocab) [inline Formula/kromium.rb:12]:
```suggestion
    system "go", "build", *std_go_args
```

File: Formula/kromium.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  end
+
+  test do
+    assert_match version.to_s(), shell_output("#{bin}/kromium -version | cut -d\":\" -f2")
```
Comment (carlocab) [inline Formula/kromium.rb:16]:
```suggestion
    assert_match version.to_s, shell_output("#{bin}/kromium -version")
```

File: Formula/kromium.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    system "mkdir", "src"
+    system "mkdir", "dst"
+    system "mkdir", "state"
```
Comment (carlocab) [inline Formula/kromium.rb:19]:
```suggestion
    %w[src dst state].map { |d| (testpath/d).mkpath }
```

File: Formula/kromium.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    system "echo hello > src/file1"
+    assert_match "file1", shell_output("ls src")
+    assert_match "", shell_output("ls dst")
```
Comment (carlocab) [inline Formula/kromium.rb:22]:
```suggestion
    (testpath/"src/file1").write "hello\n"
```

File: Formula/kromium.rb [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    system "echo { > test.cue"
+    system "echo \"SourceBucket: \\\"file://`pwd`/src\\\"\" >> test.cue"
+    system "echo \"DestinationBucket: \\\"file://`pwd`/dst\\\"\" >> test.cue"
+    system "echo \"StateBucket: \\\"file://`pwd`/state\\\"\" >> test.cue"
+    system "echo 'Transforms: [{ Type: \"Identity\" }]' >> test.cue"
+    system "echo } >> test.cue"
```
Comment (carlocab) [inline Formula/kromium.rb:28]:
```suggestion
    (testpath/"test.cue").write <<~EOS
      {
      SourceBucket: "file://`pwd`/src"
      DestinationBucket: "file://`pwd`/dst"
      StateBucket: "file://`pwd`/state"
      Transforms: [{ Type: "Identity" }]
      }
    EOS
```

File: Formula/kromium.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    system "#{bin}/kromium", "-run", "test.cue", "-render=false"
+    assert_match "file1", shell_output("ls dst")
```
Comment (carlocab) [inline Formula/kromium.rb:30]:
```suggestion
    assert_predicate testpath/"dst/file1", :exist?
```

File: Formula/kromium.rb [hunk 7]
```diff
@@ -17,14 +17,16 @@ def install
...
     assert_match "", shell_output("ls dst")
+    dir = Dir.getwd
     (testpath/"test.cue").write <<~EOS
       {
-      SourceBucket: "file://`pwd`/src"
-      DestinationBucket: "file://`pwd`/dst"
-      StateBucket: "file://`pwd`/state"
+      SourceBucket: "file://#{dir}/src"
+      DestinationBucket: "file://#{dir}/dst"
+      StateBucket: "file://#{dir}/state"
```
Comment (carlocab) [inline Formula/kromium.rb:23]:
```suggestion
      SourceBucket: "file://#{testpath}/src"
      DestinationBucket: "file://#{testpath}/dst"
      StateBucket: "file://#{testpath}/state"
```

File: Formula/kromium.rb [hunk 8]
```diff
@@ -17,14 +17,16 @@ def install
     system "echo hello > src/file1"
     assert_match "file1", shell_output("ls src")
     assert_match "", shell_output("ls dst")
+    dir = Dir.getwd
```
Comment (carlocab) [inline Formula/kromium.rb:18]:
```suggestion
```

File: Formula/kromium.rb [hunk 9]
```diff
@@ -17,14 +17,16 @@ def install
...
-      SourceBucket: "file://`pwd`/src"
-      DestinationBucket: "file://`pwd`/dst"
-      StateBucket: "file://`pwd`/state"
+      SourceBucket: "file://#{dir}/src"
+      DestinationBucket: "file://#{dir}/dst"
+      StateBucket: "file://#{dir}/state"
       Transforms: [{ Type: "Identity" }]
       }
     EOS
+    system "cp", "test.cue", "/tmp/"
```
Comment (carlocab) [inline Formula/kromium.rb:27]: Why is this needed?

**Review by carlocab — CHANGES_REQUESTED** [review]

File: Formula/kromium.rb (hunk 9 above)
Comment (sharvanath) [reply Formula/kromium.rb]: removed was testing with this. I really appreciate your adept and quick review on this 🙏

**Comment by carlocab** [comment]
This looks like a nice piece of software, but it currently does not meet the requirements for inclusion in homebrew/core. The CI log reads
```
  kromium:
    * GitHub repository not notable enough (<30 forks, <30 watchers and <75 stars)
  Error: 1 problem in 1 formula detected
```

If you wish, you can host this formula in your own tap instead. Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

Thanks for the contribution anyway!
